### PR TITLE
Adding GET endpoint for UserEntity

### PR DESCRIPTION
**Pull request by arivepr**
Adresses [this JIRA card.](https://issues.redhat.com/browse/RHCLOUD-20628)

### CODE REVIEW

**Comment by arivepr** [comment]
So dismissed updating the `last_login` with the Get endpoint for this PR since I wasn't sure if by simply hitting the Get endpoint we know that the user login.

I can append another endpoint that specifically addresses this by doing something like `getUserForLogin` or whatever, and just add the correct timestamp there. Not sure though what the best approach is there though 😅

File: rest/routes/identity.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+func GetUserIdentity(w http.ResponseWriter, r *http.Request) {
+  user := r.Context().Value(util.USER_CTX_KEY).(models.UserIdentity) 
+  userID := user.ID
+  response := make(map[string]models.UserIdentity)
+  
+  userData, err := service.GetUserIdentityByID(userID)
+  if err != nil {
+    panic(err)
+  }
+  response["data"] = userData
```
Comment (Hyperkid123) [inline rest/routes/identity.go:22]: The response should just return the `userData` directly. If a response is not a list, we just return the entity.

File: rest/routes/identity.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "github.com/RedHatInsights/chrome-service-backend/rest/models"
+  "github.com/go-chi/chi/v5"
+)
+
+func GetUserIdentity(w http.ResponseWriter, r *http.Request) {
+  user := r.Context().Value(util.USER_CTX_KEY).(models.UserIdentity) 
+  userID := user.ID
+  response := make(map[string]models.UserIdentity)
+  
+  userData, err := service.GetUserIdentityByID(userID)
```
Comment (Hyperkid123) [inline rest/routes/identity.go:18]: The context under `util.USER_CTX_KEY` actually gives you the user identity already. It does a `FirstOrCreate` lookup, that will either retrieve the entity or creates it. The logic is in the `InjectUser` middleware.

Essentially you are doing the same query twice.

File: rest/routes/identity.go (hunk 1 above)
Comment (arivepr) [reply rest/routes/identity.go:22]: Just changed it on my end, reached out to ya about the other item 😎

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**